### PR TITLE
fixes compile and linking errors for gcc 10 and clang 10

### PR DESCRIPTION
**Pull request by dhinf**
- added overloaded assignment operator for nullptr in GlobPtr
- declared two variables extern in dart base to avoid linking errors

### CODE REVIEW

**Comment by bertwesarg** [comment]
LGTM

File: dash/include/dash/GlobPtr.h [hunk 1]
```diff
@@ -121,6 +121,15 @@ class GlobPtr {
    */
   self_t &operator=(const self_t &rhs) = default;
 
+  /**
+   * Assignment operator.
```
Comment (bertwesarg) [inline dash/include/dash/GlobPtr.h]:
```suggestion
   * nullptr assignment operator.
```

**Review by devreal — APPROVED** [review]
LGTM, thanks for the fix (I thought along the same way but had to leave for a long weekend before I was able to implement it).

Also, good catch for the DART variables. One of these things where you wonder how that ever actually compiled...

**Comment by bertwesarg** [comment]
> Also, good catch for the DART variables. One of these things where you wonder how that ever actually compiled...

They were `common` symbols. A quite old concept for variables. And they get axed recently in new compilers/linkers:

* [GCC](https://gcc.gnu.org/gcc-10/porting_to.html)
* [LLVM/Clang](https://reviews.llvm.org/D75056), not yet released?
* [ARM Clang](http://www.keil.com/support/man/docs/armclang_ref/armclang_ref_chr1422532346348.htm)

**Pull request merged**